api/v1alpha3/vspheremachine_webhook.go

```diff
@@ -25,3 +32,68 @@ func (r *VSphereMachine) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		For(r).
 		Complete()
 }
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-vspheremachine,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=vspheremachines,versions=v1alpha3,name=validation.vspheremachine.infrastructure.x-k8s.io,sideEffects=None
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *VSphereMachine) ValidateCreate() error {
+	var allErrs field.ErrorList
+	spec := r.Spec
+
+	if spec.Template == "" {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "template"), "cannot be empty"))
+	}
+
+	if spec.Network.PreferredAPIServerCIDR != "" {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "PreferredAPIServerCIDR"), "cannot be set, as it will be removed and is no longer used"))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "PreferredAPIServerCIDR"), "cannot be set, as it will be removed and is no longer used"))
```